runtime/src/bank.rs

```diff
@@ -3805,7 +3804,7 @@ impl Bank {
             .stakes_cache
             .stakes()
             .highest_staked_node()
-            .unwrap_or_default();
```

Comment:
Why this change?

Reply:
The default key is the system program key which is owned by the native loader and so cannot be used as a collector id

Reply:
Oh right. Can you please add a comment explaining? Or should this fallback maybe be `incinerator::id()`

Reply:
Yeah that's unfortunate about `cfg(test)` not propagating, I forgot about that. I'll add a comment and keep it as a new unique pubkey. It's not possible to set the incinerator as the collector id in normal cluster operation so I'd rather not use that here even for tests.

Reply:
> It's not possible to set the incinerator as the collector id in normal cluster operation

Sorry if I'm missing something obvious, but why is that?

Reply:
Collector ID needs to sign to be set as the node id for a vote account and that node id is what's used as the collector id